setup.py

```diff
@@ -0,0 +1,190 @@
+    EXTRA_KWARGS = {}
+
+try:
+    import numpy as np
+except ImportError as e:
+    raise ImportError("numpy is required at installation") from e
+
+#from Cython.Build import cythonize
+#from Cython.Distutils import build_ext
+
+# all information about QuTiP goes here
+MAJOR = 0
+MINOR = 0
+MICRO = 1
+ISRELEASED = False
+VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
+REQUIRES = ['numpy (>=1.12)', 'scipy (>=1.0)', 'qutip (>=4.5)']
+EXTRAS_REQUIRE = {'graphics': ['matplotlib(>=1.2.1)']}
+INSTALL_REQUIRES = ['numpy>=1.12', 'scipy>=1.0', 'qutip>=4.5']
+PACKAGES = ['src/qip', 'src/qip/device', 'src/qip/operations', 'src/qip/compiler',
+            'src/qip/algorithms']
+PACKAGE_DATA = {}
+
+# If we're missing numpy, exclude import directories until we can
+# figure them out properly.
+INCLUDE_DIRS = [np.get_include()] if np is not None else []
+NAME = "qutip_qip"
+AUTHOR = ("Alexander Pitchford, Paul D. Nation, Robert J. Johansson, "
+          "Chris Granade, Arne Grimsmo, Nathan Shammah, Shahnawaz Ahmed, "
+          "Neill Lambert, Eric Giguere, Boxi Li")
+AUTHOR_EMAIL = ("alex.pitchford@gmail.com, nonhermitian@gmail.com, "
+                "jrjohansson@gmail.com, cgranade@cgranade.com, "
+                "arne.grimsmo@gmail.com, nathan.shammah@gmail.com, "
+                "shahnawaz.ahmed95@gmail.com, nwlambert@gmail.com, "
+                "eric.giguere@usherbrooke.ca, etamin1201@gmail.com")
+LICENSE = "BSD"
+DESCRIPTION = DOCLINES[0]
+LONG_DESCRIPTION = "\n".join(DOCLINES[2:])
+KEYWORDS = "quantum physics dynamics"
+URL = "http://qutip.org"
+CLASSIFIERS = [_f for _f in CLASSIFIERS.split('\n') if _f]
+PLATFORMS = ["Linux", "Mac OSX", "Unix", "Windows"]
+
+
+def git_short_hash():
+    try:
+        git_str = "+" + os.popen('git log -1 --format="%h"').read().strip()
+    except:
+        git_str = ""
+    else:
+        if git_str == '+': #fixes setuptools PEP issues with versioning
+            git_str = ''
+    return git_str
+
+FULLVERSION = VERSION
+if not ISRELEASED:
+    FULLVERSION += '.dev'+str(MICRO)+git_short_hash()
+
+# NumPy's distutils reads in versions differently than
+# our fallback. To make sure that versions are added to
+# egg-info correctly, we need to add FULLVERSION to
+# EXTRA_KWARGS if NumPy wasn't imported correctly.
+if np is None:
+    EXTRA_KWARGS['version'] = FULLVERSION
+
+
+def write_version_py(filename='src/qip/version.py'):
+    cnt = """\
+# THIS FILE IS GENERATED FROM QUTIP SETUP.PY
+short_version = '%(version)s'
+version = '%(fullversion)s'
+release = %(isrelease)s
+"""
+    a = open(filename, 'w')
+    try:
+        a.write(cnt % {'version': VERSION, 'fullversion':
+                FULLVERSION, 'isrelease': str(ISRELEASED)})
+    finally:
+        a.close()
+
+local_path = os.path.dirname(os.path.abspath(sys.argv[0]))
+print(local_path)
+os.chdir(local_path)
+sys.path.insert(0, local_path)
+print(local_path)
+sys.path.insert(0, os.path.join(local_path, 'src/qip'))  # to retrive _version
+print(local_path)
```

Comment:
This whole paragraph looks like debugging code that can be removed? The `prints` should definitely no be here and I don't see `version` is only written here, not imported, so probably it doesn't need to be on `sys.path` anywhere?

Reply:
Done, thanks for the comment! I think I'd left it in there when I was working on the same PR before (now closed).